### PR TITLE
EVG-1977: cli for admin restart failed tasks

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: cli/admin.go [hunk 1]
```diff
@@ -1,12 +1,71 @@
...
 	"github.com/evergreen-ci/evergreen/rest/model"
 	"github.com/mongodb/grip"
 	"github.com/pkg/errors"
 	"golang.org/x/net/context"
 )
 
+type AdminRestartTasks struct {
+	GlobalOpts *Options `no-flag:"true"`
+	DryRun     bool     `long:"dry-run" short:"n" description:"run migration in a dry-run mode"`
+	StartTime  string   `long:"startAt" short:"s" description:"RFC339 formated date of the start time for the period of failed tasks to restart"`
```
Comment (john-m-liu) [inline cli/admin.go]: missing a 3 in RFC3339

File: cli/admin.go [hunk 2]
```diff
@@ -1,12 +1,71 @@
...
+	client, settings, err := getAPIV2Client(c.GlobalOpts)
+	if err != nil {
+		return errors.Wrap(err, "problem configuring api client")
+	}
+
+	client.SetAPIUser(settings.User)
+	client.SetAPIKey(settings.APIKey)
+
+	ctx := context.Background()
+	if err = client.RestartRecentTasks(ctx, startAt, endAt); err != nil {
```
Comment (john-m-liu) [inline cli/admin.go]: DryRun isn't used at all

File: rest/client/rest.go [hunk 3]
```diff
@@ -196,6 +197,35 @@ func (c *communicatorImpl) GetServiceFlags(ctx context.Context) (*model.APIServi
...
+		version: apiVersion2,
+		path:    "admin/restart",
+	}
+
+	payload := struct {
+		StartTime time.Time `json:"start_time"`
+		EndTime   time.Time `json:"end_time"`
+		DryRun    bool      `json:"dry_run"`
+	}{
+		DryRun:    false,
```
Comment (john-m-liu) [inline rest/client/rest.go]: I think you forgot to implement DryRun

File: cli/admin.go (hunk 1 above)
Comment (tychoish) [reply cli/admin.go]: done

File: cli/admin.go (hunk 2 above)
Comment (tychoish) [reply cli/admin.go]: I think we should, after the extra options are implemented later this sprint refactor change the API of RestartRecentTasks to be: (ctx, apimodels.RestartTaskOptions), which will let us set options like dryrun without fussing with the API too much.

**Pull request merged**